backend/routes.js

```diff
@@ -341,9 +357,9 @@ if (process.env.NODE_ENV === 'development') {
       const { hash, data } = request.payload
       if (!hash) return Boom.badRequest('missing hash')
       if (!data) return Boom.badRequest('missing data')
-      const ourHash = createCID(data)
-      if (ourHash !== hash) {
-        console.error(`hash(${hash}) != ourHash(${ourHash})`)
+      const ourHashes = Object.values(multicodes).map((multicode) => createCID(data, ((multicode: any): number)))
+      if (ourHashes.includes(hash)) {
+        console.error(`hash(${hash}) != ourHash(${ourHashes.join(',')})`)
```

Comment:
This is a very odd chunk of code here...

First off, I think the condition is backwards, should be `!ourHashes.includes(hash)`. And then still the error message would need to be updated to specify that the hash isn't in that list (as opposed to not being equal to the list).

However, I don't see a good reason to do it this way... can you update the code to extract the prefix and then based on the prefix decide which multicode to use? That way we don't need to try them all - we know immediately which one we need to use.

Reply:
You're right about the condition, seems to be wrong.